### PR TITLE
Move from tinyxml to tinyxml2 in FWCore-MessageLogger

### PR DESCRIPTION
**Pull request by mrodozov**
Recently we added tinyxml2 external and now we are moving pkgs that were using tinyxml to tinyxml2.

### CODE REVIEW

**Comment by mrodozov** [comment]
please test

File: FWCore/MessageLogger/src/JobReport.cc [hunk 1]
```diff
@@ -611,26 +620,33 @@ namespace edm {
     if(impl_->ost_) {
       std::ostream& ost = *(impl_->ost_);
       ost << "<ReadBranches>\n";
+      tinyxml2::XMLDocument doc;
       for(auto const& iBranch : impl_->readBranches_) {
-        TiXmlElement branch("Branch");
-        branch.SetAttribute("Name", iBranch.first);
-        branch.SetAttribute("ReadCount", iBranch.second);
-        ost << branch << "\n";
+	tinyxml2::XMLPrinter printer;
```
Comment (Dr15Jones) [inline FWCore/MessageLogger/src/JobReport.cc]: indentation

File: FWCore/MessageLogger/src/JobReport.cc [hunk 2]
```diff
@@ -611,26 +620,33 @@ namespace edm {
...
+	tinyxml2::XMLElement * branch = doc.NewElement("Branch");
+        branch->SetAttribute("Name", iBranch.first.c_str());
+        branch->SetAttribute("ReadCount", int64_t(iBranch.second));
+	branch->Accept(&printer);
```
Comment (Dr15Jones) [inline FWCore/MessageLogger/src/JobReport.cc]: indentation

File: FWCore/MessageLogger/src/JobReport.cc [hunk 3]
```diff
@@ -561,12 +567,15 @@ namespace edm {
...
-      fallback.SetAttribute("Lfn", lfn);
+      tinyxml2::XMLDocument doc;
+      tinyxml2::XMLPrinter printer;
+      tinyxml2::XMLElement * fallback = doc.NewElement("FallbackAttempt");
+      fallback->SetAttribute("Pfn", pfn.c_str());
+      fallback->SetAttribute("Lfn", lfn.c_str());
       {
         std::lock_guard<std::mutex> lock(write_mutex);
-        msg << fallback << "\n";
+	fallback->Accept(&printer);
```
Comment (Dr15Jones) [inline FWCore/MessageLogger/src/JobReport.cc]: indention

File: FWCore/MessageLogger/src/JobReport.cc [hunk 4]
```diff
@@ -611,26 +620,33 @@ namespace edm {
...
       for(auto const& iBranch : impl_->readBranchesSecFile_) {
-        TiXmlElement branch("Branch");
-        branch.SetAttribute("Name", iBranch.first);
-        branch.SetAttribute("ReadCount", iBranch.second);
-        ost << branch << "\n";
+	tinyxml2::XMLPrinter printer;
+	tinyxml2::XMLElement * branch = doc.NewElement("Branch");
+        branch->SetAttribute("Name", iBranch.first.c_str());
+        branch->SetAttribute("ReadCount", int64_t(iBranch.second));
+	branch->Accept(&printer);
```
Comment (Dr15Jones) [inline FWCore/MessageLogger/src/JobReport.cc]: indentation

File: FWCore/MessageLogger/src/JobReport.cc [hunk 5]
```diff
@@ -611,26 +620,33 @@ namespace edm {
...
       }
       ost << "</ReadBranches>\n";
       if(!impl_->readBranchesSecSource_.empty()) {
         ost << "<SecondarySourceReadBranches>\n";
         for(auto const& iBranch : impl_->readBranchesSecSource_) {
-          TiXmlElement branch("Branch");
-          branch.SetAttribute("Name", iBranch.first);
-          branch.SetAttribute("ReadCount", iBranch.second.value().load());
-          ost << branch << "\n";
+	  tinyxml2::XMLPrinter printer;
```
Comment (Dr15Jones) [inline FWCore/MessageLogger/src/JobReport.cc]: indentation

File: FWCore/MessageLogger/src/JobReport.cc [hunk 6]
```diff
@@ -611,26 +620,33 @@ namespace edm {
...
+	  tinyxml2::XMLElement * branch = doc.NewElement("Branch");
+          branch->SetAttribute("Name", iBranch.first.c_str());
+          branch->SetAttribute("ReadCount", int64_t(iBranch.second.value().load()));
+	  branch->Accept(&printer);
```
Comment (Dr15Jones) [inline FWCore/MessageLogger/src/JobReport.cc]: indentation

File: FWCore/MessageLogger/src/JobReport.cc [hunk 7]
```diff
@@ -611,26 +620,33 @@ namespace edm {
...
+        branch->SetAttribute("ReadCount", int64_t(iBranch.second));
+	branch->Accept(&printer);
+        ost << printer.CStr();
       }
       for(auto const& iBranch : impl_->readBranchesSecFile_) {
-        TiXmlElement branch("Branch");
-        branch.SetAttribute("Name", iBranch.first);
-        branch.SetAttribute("ReadCount", iBranch.second);
-        ost << branch << "\n";
+	tinyxml2::XMLPrinter printer;
```
Comment (Dr15Jones) [inline FWCore/MessageLogger/src/JobReport.cc]: indentation

File: FWCore/MessageLogger/src/JobReport.cc [hunk 8]
```diff
@@ -546,12 +549,15 @@ namespace edm {
...
-      skipped.SetAttribute("Lfn", lfn);
+      tinyxml2::XMLDocument doc;
+      tinyxml2::XMLPrinter printer;
+      tinyxml2::XMLElement * skipped = doc.NewElement("SkippedFile");
+      skipped->SetAttribute("Pfn", pfn.c_str());
+      skipped->SetAttribute("Lfn", lfn.c_str());
       {
         std::lock_guard<std::mutex> lock(write_mutex);
-        msg << skipped << "\n";
+	skipped->Accept(&printer);
```
Comment (Dr15Jones) [inline FWCore/MessageLogger/src/JobReport.cc]: Please keep consistent indentation

File: FWCore/MessageLogger/src/JobReport.cc [hunk 9]
```diff
@@ -725,7 +742,7 @@ namespace edm {
   JobReport::dumpFiles(void) {
     std::ostringstream msg;
 
-
+    tinyxml2::XMLDocument doc;
```
Comment (Dr15Jones) [inline FWCore/MessageLogger/src/JobReport.cc]: Will having the `doc` outside of the loop mean it will accumulate memory until the loop finishes?

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
The indentation is fixed.

File: FWCore/MessageLogger/src/JobReport.cc (hunk 9 above)
Comment (mrodozov) [reply FWCore/MessageLogger/src/JobReport.cc]: The last commit is using a method to delete the document nodes when the doc is used to create new text objects inside loops.

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

**Comment by efeyazgan** [comment]
@intrepid42 could you please take a look?

**Comment by alberto-sanchez** [comment]
Anything here?

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
@alberto-sanchez @intrepid42 @xjanssen in my understanding RivetHarvesting should merge together multiple rivet files. So the test should be mimicking such a simple workflow and verify that the output is meaningful. Could we please try to move forward?

**Comment by xjanssen** [comment]
Hi, we don't use that plugin ourself and we don't think someone is.

**Comment by mrodozov** [comment]
I'd like to ask @mkirsano and @guitargeek if they can clarify as they made the most recent changes.

**Comment by guitargeek** [comment]
Sorry I have no clue, I was only touching that file to fix memory errors.

**Comment by fabiocos** [comment]
@efeyazgan @alberto-sanchez @intrepid42 @xjanssen @qliphy so why don't we just remove the class from the release, if nobody uses it and nobody knows how to test it? There is no point in wasting people's effort on something abandoned and unmaintained

**Comment by efeyazgan** [comment]
@lenzip informed us that it could be removed.

**Comment by fabiocos** [comment]
@efeyazgan @lenzip doing and undoing... Fine, could you please provide a PR that removes the module?

**Comment by efeyazgan** [comment]
@fabiocos, this is something from 2013 and not fully developed. 

The PR is #24810.

**Comment by lenzip** [comment]
Hi @fabiocos, @efeyazgan. That class was part of an attempt that I made long ago at using Rivet analyses for release validation purposes. The  attempt involved two steps: 1) create ME from Rivet internal histograms, 2) call whatever was done in the analyses finalize method in a DQM harvesting module, which is the class in question. This never really worked. Although it might be still interesting to pursue, if that class hurts in some way, I think we can remove it, unless someone is interested in reviving the idea. I can discuss with interested parties the problems I had.

**Comment by fabiocos** [comment]
@lenzip I recall the problem, I guess that most of the people is using the (more or less) native scripts provided by the Rivet authors anyway. The point is that maintaining a class that is unused is anyway costly, and if nobody even know how to validate it better to take it out. If at some point someone is really interested in revamping the effort and finalize it we can always add a new version in the future

**Comment by alberto-sanchez** [comment]
please test with #24810

**Comment by fabiocos** [comment]
@qliphy @efeyazgan @alberto-sanchez ping

**Comment by alberto-sanchez** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**